### PR TITLE
[NFC][LLVM][SVE][ISel] Remove redundant type information from Pat targets.

### PR DESCRIPTION
**Pull request by paulwalker-arm**

### CODE REVIEW

File: llvm/lib/Target/AArch64/AArch64SVEInstrInfo.td [hunk 1]
```diff
@@ -2521,9 +2517,9 @@ let Predicates = [HasSVEorSME] in {
               (ADDVL_XXI GPR64:$op, $imm)>;
 
     def : Pat<(add GPR32:$op, (i32 (trunc (vscale (sve_rdvl_imm i32:$imm))))),
-              (i32 (EXTRACT_SUBREG (ADDVL_XXI (INSERT_SUBREG (i64 (IMPLICIT_DEF)),
-                                             GPR32:$op, sub_32), $imm),
-                                   sub_32))>;
+              (EXTRACT_SUBREG (ADDVL_XXI (INSERT_SUBREG (i64 (IMPLICIT_DEF)),
```
Comment (kmclaughlin-arm) [inline llvm/lib/Target/AArch64/AArch64SVEInstrInfo.td:2520]: Can the i64 be removed for `IMPLICIT_DEF` here as it was for the pattern above?

Comment (paulwalker-arm) [reply llvm/lib/Target/AArch64/AArch64SVEInstrInfo.td]: Yes. I've pushed a commit to replace them.

**Review by kmclaughlin-arm — APPROVED** [review]
LGTM!

**Comment by aemerson** [comment]
Is there a reason why these need to be removed? When we eventually bring up GlobalISel for SVE, we may want these additional type information for the importer. We see cases today of some NEON patterns being importable once the types are specified.

I'm not blocking this since it's not even certain that these patterns will be problematic for the importer.

**Comment by paulwalker-arm** [comment]
> Is there a reason why these need to be removed? 

Just consistency.  Most all the patterns (well those related to SVE at least) don't include this information so these few are the outliers.

**Pull request merged**